### PR TITLE
Add abiDecode wrapper in CompilerUtils

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

**Comment by axic** [comment]
Had this around for a while.

**Comment by chriseth** [comment]
I think this is superseded by a PR waiting for your review for quite a while now: https://github.com/ethereum/solidity/pull/3308/files#diff-ee617ea709ab57f19fa067e365a9b4f8R1860

**Comment by axic** [comment]
I know I've seen that was my trigger to push this. Will see if it can/need to be reconciled.

**Comment by chriseth** [comment]
The other version has a length check, so I think that is superior.

**Comment by axic** [comment]
@chriseth I merged the cleanup one from the other PR here so the documentation is kept. Can you double check it is valid?

Also this one uses it in one more place (in `appendExternalCall`) compared to the other one.

File: libsolidity/codegen/CompilerUtils.cpp [hunk 1]
```diff
@@ -206,7 +206,11 @@ void CompilerUtils::abiDecode(TypePointers const& _typeParameters, bool _fromMem
 		}
 		// stack: v1 v2 ... v(k-1) v(k) base_offset mem_offset
```
Comment (axic) [inline libsolidity/codegen/CompilerUtils.cpp]: @chriseth I assume `mem_offset` is the point until we have written something into memory?

Comment (chriseth) [reply libsolidity/codegen/CompilerUtils.cpp]: No, `base_offset` is the position of the beginning of the current "block" (the point from which on data pointers are taken) and `mem_offset` is the point inside the current block where we are currently reading.

Comment (axic) [reply libsolidity/codegen/CompilerUtils.cpp]: Do we have the pointer in memory or is the free memory pointer updated properly?

**Comment by chriseth** [comment]
I'm pretty sure this is redundant now. If not, please reopen and rebase.

**Pull request closed without merging**